src/Illuminate/Support/Collection.php

```diff
@@ -768,6 +768,16 @@ public function toJson($options = 0)
 	}
 
 	/**
+	 * Get an hash of the collection.
+	 *
+	 * @return string
+	 */
+	public function hash()
+	{
+		return md5($this);
```

Comment:
Sure. Will sha1 behave correctly? I normally use it over md5 due to it's much reduced collision probability.